enhancements/cluster-scope-secret-volumes/shared-resources-via-openshift-builds-operator.md

```diff
@@ -0,0 +1,291 @@
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+As noted above, the operator will not deploy the Shared Resource CSI Driver if the cluster has
+the respective OCP feature gate enabled. This generally applies to tech preview clusters or
+clusters with custom fature gate tunings.
+
+With the start of OCP 4.16, the Shared Resource CSI driver operator will be removed from the
+[Cluster Storage Operator](https://github.com/openshift/cluster-storage-operator). The feature
+will likewise be removed from OpenShift's feature sets - any code that needs the most up to date
+set of feature sets will need to be updated (ex - openshift/library-go).
+The Shared Resource CSI driver itself will need to be removed from the OpenShift payload. Its
+components will need to be re-productized as an OLM operand.
+
+Notably, the design of this operator does **_not_** intend to use OLM API dependency resolution
+to automate the deployment and management of OpenShift Pipelines. Cluster admimins will need to
+separately (and manually) install OpenShift Pipelines.
+
+
+### Risks and Mitigations
+
+#### Duplicate CSI Drivers
+
+On a tech preview cluster, the Same CSI driver could be deployed twice on tech preview clusters. The operator will check if the `CSIDriverSharedResource` feature is enabled before
+deploying the Shared Resource CSI driver itself.
+
+
+#### Unwanted components/CRDs
+
+Admins or users may want to use the Shared Resource CSI Driver without Shipwright or Tekton - for
+example, developers or teams that heavily rely on the `BuildConfig` APIs today. To address this
+concern, the OpenShift Builds operator will not rely on OLM API resolution to automate the
+installation and deployment of OpenShift Pipelines. The operator will instead report an appropriate
+status message if `spec.shipwright.build.enabled` is set to `true` and it detects that OpenShift
+Pipelines has not been installed.
+
+
+### Drawbacks
+
+- Shared resources no longer a part of OpenShift.
+- Customers who need entitlements in builds will need to install a new operator, or resort to
+  current cumbersome work-arounds.
+- Potential impact on Red Hat components that need shared resources and/or RHEL entitlements.
+- Manual installation of OpenShift Pipelines is required to deploy Shipwright Builds. Shipwright's
+  upstream operator currently installs Tekton via [OLM API dependency resolution](https://olm.operatorframework.io/docs/concepts/olm-architecture/dependency-resolution/).
+  This proposal will deviate from the "upstream" behavior to support customers who want the Shared
+  Resource CSI Driver, but are not interested in Shipwright or Tekton.
+
+
+## Design Details
+
+
+### Test Plan
+
+- Shared resource logic in the OCP Builds tech preview suite will need to be migrated to the
+  OpenShift Builds operator test suites
+- OpenShift Builds operator will need a techpreview variant to ensure the shared resource CSI
```

Comment:
i'm not sure what this means or why it would be needed? 

maybe you mean "needs a test that runs in the techpreview variant to....."?

i'll reiterate here that i think it might be better if the operator directly looked for evidence that the CSI driver was already installed/enabled.  (Speaking of, i assume this is a cluster scoped operator only? because if not we need to talk about what happens if it gets installed multiple times, also)

and still need coverage for what happens when "builds operator is installed + shared resources enabled *before* someone turns on techpreview on the cluster"

Reply:
I think different names for the CSI driver obj itself and a new namepace for the daemonset / webhooks would be sufficient for making sure the new operator only manages the GA version, and we leave the tech preview version to the OCP cluster storage operator based stuff 

Or do you all think I am missing something @bparees @adambkaplan

Reply:
i generally like that approach, again keeping in mind that the existing buildconfig integrations that support the TP csi driver might need updating/revising.

Reply:
ack, thanks

Reply:
Updated with new direction on CSI driver naming